### PR TITLE
Format and lint JS SDK code

### PR DESCRIPTION
**Pull request by bafonins**
#### Summary

This PR adds code linting and formatting to the js sdk.

#### Changes

- Add `prettier` 
- Add `eslint`
- Format js sdk code with prettier
- Add `jsSDK:lint` and `jsSDK:fmt` mage targets
- Include linting js sdk during ci

#### Notes for Reviewers

@htdvisser and @rvolosatovs  as always, only the `mage/ci` parts

### CODE REVIEW

File: .mage/js_sdk.go [hunk 1]
```diff
@@ -90,6 +90,36 @@ func (k JsSDK) Watch() error {
...
+
+	if mg.Verbose() {
+		fmt.Println("Running prettier on sdk .js files")
+	}
+	yarn, err := k.yarn()
+	if err != nil {
+		return err
+	}
+
+	return yarn("run", "fmt")
```
Comment (rvolosatovs) [inline .mage/js_sdk.go]: Why not define `func (k JsSDK) runYarn(string) error` and do
```go
return k.yarnRun(<cmd>) // <cmd> would be "fmt" here, for example
```

instead of the whole
```go
	yarn, err := k.yarn()
	if err != nil {
		return err
	}

	return yarn("run", <cmd>)
```
block in most of these methods?

File: .travis.yml [hunk 2]
```diff
@@ -9,137 +9,137 @@ go: 1.12.x
...
-  - NODE_ENV=production
-  - YARN_CACHE_FOLDER=$HOME/.cache/yarn
-  - MAGE=$HOME/.cache/mage/mage
-  - TEST_SLOWDOWN=8
-  - TEST_REDIS=1
-  - PATH=/snap/bin:$PATH
-  - HUGO_BASE_URL=https://thethingsnetwork.github.io/lorawan-stack/
-  - DOCKER_IMAGE=thethingsnetwork/lorawan-stack
-  - DOCKER_IMAGE_DEV=thethingsnetwork/lorawan-stack-dev
+    - GOPROXY=https://proxy.golang.org
```
Comment (rvolosatovs) [inline .travis.yml]: unrelated?

File: .travis.yml [hunk 3]
```diff
@@ -9,137 +9,137 @@ go: 1.12.x
...
 services:
-- docker
+  - docker
 cache:
   directories:
-  - "$GOPATH/pkg/mod"
-  - "$HOME/.cache/go-build"
-  - "$HOME/.cache/mage"
-  - "$HOME/.cache/yarn"
+    - '$GOPATH/pkg/mod'
```
Comment (rvolosatovs) [inline .travis.yml]: Why?

File: .mage/js_sdk.go (hunk 1 above)
Comment (bafonins) [reply .mage/js_sdk.go]: not in this pr. we will revisit this part when publishing the sdk to the npm registry.

Comment (rvolosatovs) [reply .mage/js_sdk.go]: Can you add a TODO with an issue reference?

Comment (bafonins) [reply .mage/js_sdk.go]: no. your proposed improvement has nothing to with this specific pr. `Lint()` and `Fmt()` follow the structure that was implemented before.

see https://github.com/TheThingsNetwork/lorawan-stack/issues/30#issuecomment-522974595
